File: String Matching/Q2. Rotate String/rotateString.js
/**
 * @param {string} s
 * @param {string} goal
 * @return {boolean}
 */
var rotateString = function (s, goal) {
    const len = s.length
    let idx = 0

    while (idx < len) {
        let rev = s.substring(idx) + s.substring(0, idx)
        if (rev === goal) {
            return true
        }
        idx++
    }
    return false
};

File: String Matching/Q2. Rotate String/rotateString.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    bool rotateString(string s, string goal) {
        int sz = s.size();
        int idx = 0;

        while(idx < sz){
            string rev = s.substr(idx) + s.substr(0,idx);

            if(rev == goal)return true;
            idx++;
        }
        return false;
    }
};